# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1753899080 -7200
# Parent  49038a65ba7c6591af87ff8268c194dbd4ca6fbf
Message Rethreading, introduce pref mail.thread.rethreading.allowDragAndDrop.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -759,8 +759,9 @@ pref("font.name-list.monospace.el", "mon
 
 pref("mail.setup.loglevel", "Warn");
 
 // Message Editor/Rethreading.
+pref("mail.thread.rethreading.allowDragAndDrop", true);
 // Editable headers. Formatted in Train-Case for display purposes to
 // accomodate header names that are not among the common localizable ones.
 // The default may be changed, eg: '["Date","From","In-Reply-To"]' and adding
 // a "*" anywhere will list all headers in the message.
diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -5354,8 +5354,12 @@ var threadPane = {
       event.preventDefault();
       return;
     }
 
+    this._rethreadingAllowDragAndDrop = Services.prefs.getBoolPref(
+      "mail.thread.rethreading.allowDragAndDrop",
+      true
+    );
     if (threadTree.inKeyDnD) {
       this._onKeyDragEnd(event);
     }
     threadTree.currentIndex = row.index;
@@ -5620,8 +5624,11 @@ var threadPane = {
    * @param {integer} dragIndex - Index of message to rethread.
    * @param {integer} dropIndex - Index of potential new parent.
    */
   _canReThread(dragIndex, dropIndex) {
+    if (!this._rethreadingAllowDragAndDrop) {
+      return false;
+    }
     if (!gViewWrapper.showThreaded || threadTree.selectedIndices.length > 1) {
       return false;
     }
     // Preliminary test; use nsMsgViewIndex_None as params.
